revise getting started and cli docs (#53)
GoogleChrome/lighthouse-ci: docs/getting-started.md

## Prerequisites

Your project should meet the following requirements:

1. Source code is managed with git (GitHub, GitLab, Bitbucket, etc).
2. Branches/pull requests are gated on the results of a continuous integration build process (Travis CI, CircleCI, Azure Pipelines, AppVeyor, GitHub Actions, etc).
3. Your CI process can build your project into production assets (typically provided as an `npm run build` command by most frameworks).

> 4. Your project either A) has a command that runs a production-equivalent web server, or B) is a static site.

Reviewer: ```suggestion
4. Your project either:
    A) has a command that runs a production-equivalent web server.
    B) is a static site.
```

In the examples that follow, use of GitHub and Travis CI are assumed but the same concepts apply to other providers. Refer to your specific provider's documentation for how to accomplish each setup step.

## Quick Start

Lighthouse CI comes with an automatic pipeline, called `autorun`, that should work for many web projects. If you have any complicated moving parts, `autorun` might not work for you. Read up on the details in [the full setup guide](#setup).

**.travis.yml**

```yaml
# xenial (the default) or newer required
language: node_js
node_js:
  - 10 # Node 10 LTS or later required
addons:
  chrome: stable # make sure to have Chrome available
before_install:
  - npm install -g @lhci/cli@next # install LHCI
script:
  - npm run build # build your site
  - npm test      # run your normal tests
  - lhci autorun  # run lighthouse CI against your static site
```

That's it! With this in place, you'll have Lighthouse reports collected and and asserted.

> See the [autorun docs](./cli#autorun) for more including running your own webserver and uploading the reports for later viewing.

Reviewer: good call, I didn't like how complicated the quick start was getting :)

# ...
```

#### Serve Your Site

To run Lighthouse CI, the built site you'd like to test with Lighthouse needs to be available on a server. You can either use the built-in Lighthouse CI static server, a custom local development server, or deploy to a staging location that's accessible to your CI.

> For this example, we'll assume your site is already built in a local directory called `./dist`, and we'll use the the `http-server` node package as an example custom server implementation (do not follow this structure just to use `http-server`, it is less fully featured than `lhci collect --static-dist-dir=./dist`, see [Run Lighthouse CI](#run-lhci) for more).

Reviewer: ```suggestion
For this example, we'll assume your site is already built in a local directory called `./dist`, and we'll use the the `http-server` node package as an example custom server implementation (do not follow this structure just to use `http-server`, it is less fully featured than `lhci collect --static-dist-dir=./dist`, see [Run Lighthouse CI](#run-lighthouse-ci) for more).
```

}
```

```bash
lhci assert --rc-file=lighthouserc.json
EXIT_CODE=$?
```

## Storing Reports

You have two options for storing the collected Lighthouse reports for later viewing.

### Saving Reports to Temporary Public Storage

While the Lighthouse CI server offers the complete LHCI experience, you do have to host it yourself. If you'd like basic storage so you can just get a URL for each report, we offer a free service that provides temporary public storage.

> **NOTE: as the name implies this is _temporary_ and _public_ storage, do not use if you're uncomfortable with the idea of your Lighthouse reports being stored on a public URL on Google Cloud Infrastructure. Reports are automatically deleted 7 days after upload.**

Reviewer: ```suggestion
**NOTE: as the name implies this is _temporary_ and _public_ storage, do not use if you're uncomfortable with the idea of your Lighthouse reports being stored on a public URL on Google Cloud infrastructure. Reports are automatically deleted 7 days after upload.**
```

```bash
# If you use `lhci assert`, upload must follow after it.
lhci upload --target=temporary-public-storage
```

If you're using `autorun`, setting `ci.upload.target` to `temporary-public-storage` in your `lighthouserc.json` is all that's necessary. The upload step will happen automatically.

### The Lighthouse CI Server

<img src="https://user-images.githubusercontent.com/39191/68522781-496bad00-0264-11ea-800a-ed86dbb04366.png" width="48%"> <img src="https://user-images.githubusercontent.com/39191/68522269-7917b680-025e-11ea-8d96-2774c0a0b04c.png" width="48%">

Historical reports and advanced report diffing is available with the Lighthouse CI server. For setup of the server itself, see [our server recipe](./recipes/README.md).

Once the server is set up, create a a new project with the lhci wizard:

```bash
$ lhci wizard
? Which wizard do you want to run? new-project
? Which server would you like to use? http://url-of-your-lhci-server.com/
? What would you like to name the project? My Favorite Project

> ? Where is the project built? https://travis-ci.org/GoogleChrome/lighthouse-ci

Reviewer: ```suggestion
? Where is the project's code hosted? https://github.com/GoogleChrome/lighthouse-ci
```

Author: oh really? i thought this question was asking about your travis URL.

oh you changed it. neat :)

Created project My Favorite Project (XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX)!
Use token XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX to connect.
```

> Add your token to your CI with the environment variable `LHCI_TOKEN`. Alternatively, you can pass it with to `upload` with the `--token` flag:

Reviewer: ```suggestion
Add your token to your CI with the environment variable `LHCI_TOKEN`. Alternatively, you can pass it to `upload` with the `--token` flag:
```
